src/main/java/it/pagopa/pagopa/apiconfig/service/EncodingsService.java

```diff
@@ -51,6 +51,16 @@ public Encoding createCreditorInstitutionEncoding(@NotNull String creditorInstit
         if (codifichePaRepository.findByCodicePaAndFkPa_ObjId(encoding.getEncodingCode(), pa.getObjId()).isPresent()) {
             throw new AppException(AppError.ENCODING_CREDITOR_INSTITUTION_CONFLICT, encoding.getEncodingCode(), creditorInstitutionCode);
         }
+
+        // check encoding code length
+        // QR-CODE must be 11 digits
+        // BARCODE-128-AIM must be 12 digits
+        if ((encoding.getCodeType().equals(Encoding.CodeTypeEnum.QR_CODE) && encoding.getEncodingCode().length() != 11) ||
```

Comment:
can we use `@Size` annotation instead?

Reply:
no, because it depends on an another field